edx/edx-enterprise
# ENT-263: Submit analytics events on data sharing consent form

**Description:** This pull request adds segment.io events (when available) to the new data sharing consent form created in #68.

**JIRA:** [https://openedx.atlassian.net/browse/ENT-263](ENT-263)

**Dependencies:** Depends on #68 (please review from commit 78ebef0)

**Merge deadline:** 27 March 2017

**Installation instructions:**

1. Standard Enterprise installation; no requirements beyond the basic app
2. Requires a segment.io developer account (free)
3. Set `SEGMENT_KEY` in `lms.auth.json` to the "write key" for your single free source in segment.io
4. Create an EnterpriseCustomer which requires data sharing consent at enrollment

**Testing instructions:**

1. In your EnterpriseCustomer, link a user and enroll that user in a course.
2. In an incognito window, log into that user, and attempt to browse to the course. Note the data sharing consent UI that shows up.
3. In the segment.io debugger, observe a `data_consent_form_shown` event.
4. Decline to provide consent. Observe that you are redirected to the Dashboard.
5. In the segment.io debugger, observe a `data_consent_denied` event.
6. Attempt to browse to the course. Again, note that the data sharing consent UI shows up.
7. This time, check the box to consent, and submit the form.
8. In the segment.io debugger, observe a `data_consent_accepted` event.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:**

* Jasmine tests coming soon
* This depends on #68; perform code review starting at commit 78ebef0
